test/extended/image_ecosystem/mongodb_replica_petset.go

```diff
@@ -0,0 +1,112 @@
+package image_ecosystem
+
+import (
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	dbutil "github.com/openshift/origin/test/extended/util/db"
+)
+
+var _ = g.Describe("[image_ecosystem][mongodb][Slow] openshift mongodb replication (with petset)", func() {
+	defer g.GinkgoRecover()
+
+	const templatePath = "https://raw.githubusercontent.com/sclorg/mongodb-container/master/examples/petset/mongodb-petset-persistent.yaml"
+
+	oc := exutil.NewCLI("mongodb-petset-replica", exutil.KubeConfigPath()).Verbose()
+
+	g.Describe("creating from a template", func() {
+		g.It(fmt.Sprintf("should process and create the %q template", templatePath), func() {
+			oc.SetOutputDir(exutil.TestContext.OutputDir)
+
+			g.By("creating persistent volumes")
+			_, err := exutil.SetupHostPathVolumes(
+				oc.AdminKubeREST().PersistentVolumes(),
+				oc.Namespace(),
+				"256Mi",
+				3,
+			)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			defer func() {
+				// We're removing only PVs because all other things will be removed
+				// together with namespace.
+				// TODO: unfortunately it doesn't remove the directories on the filesystem.
+				// Need to figure out how to fix it.
+				err := exutil.CleanupHostPathVolumes(oc.AdminKubeREST().PersistentVolumes(), oc.Namespace())
```

Comment:
Maybe it's because I'm working in the VM:
```console
[vagrant@localhost origin]$ ls -ld /tmp/openshift-extended-tests/persistent-volumes668468432/0000972706799/
drwxrwxrwx. 4 vagrant vagrant 80 Oct 20 20:07 /tmp/openshift-extended-tests/persistent-volumes668468432/0000972706799/
[vagrant@localhost origin]$ ls -l /tmp/openshift-extended-tests/persistent-volumes668468432/0000972706799/
total 0
drwxr-xr-x. 2 1000040000 root  80 Oct 20 17:03 diagnostic.data
drwxr-xr-x. 2 1000040000 root 100 Oct 20 17:03 journal
[vagrant@localhost origin]$ rm -rf /tmp/openshift-extended-tests/persistent-volumes668468432/0000972706799/
rm: cannot remove ‘/tmp/openshift-extended-tests/persistent-volumes668468432/0000972706799/diagnostic.data/metrics.2016-10-20T17-03-42Z-00000’: Permission denied
rm: cannot remove ‘/tmp/openshift-extended-tests/persistent-volumes668468432/0000972706799/diagnostic.data/metrics.2016-10-20T17-03-06Z-00000’: Permission denied
rm: cannot remove ‘/tmp/openshift-extended-tests/persistent-volumes668468432/0000972706799/journal/WiredTigerPreplog.0000000002’: Permission denied
rm: cannot remove ‘/tmp/openshift-extended-tests/persistent-volumes668468432/0000972706799/journal/WiredTigerPreplog.0000000001’: Permission denied
rm: cannot remove ‘/tmp/openshift-extended-tests/persistent-volumes668468432/0000972706799/journal/WiredTigerLog.0000000002’: Permission denied
```